site/content/blog/multi-env-deployment-ytt-kapp.md

```diff
@@ -0,0 +1,331 @@
+---
+title: "Multi environment deployment with ytt and kapp"
+slug: multi-env-deployment-ytt-kapp
+date: 2022-03-10
+author: Yash Sethiya
+excerpt: "Handle multiple environments with ytt and kapp"
+image: /img/logo.svg
+tags: ['carvel', 'kapp', 'ytt']
+
+---
+
+One of the most typical challenges when deploying a complex application is the handling of different deployment environments during the software lifecycle.
+
+The most typical setup is the trilogy of QA/Staging/Production environments. An application developer needs an easy way to deploy to the different environments and also to understand what version is deployed where.
+
+Unlike [many other tools used for templating](https://carvel.dev/ytt/docs/v0.40.0/ytt-vs-x/), ytt takes a different approach to work with YAML files. Instead of interpreting YAML configuration as plain text, it works with YAML structures such as maps, lists, YAML documents, scalars, etc. By doing so ytt is able to eliminate a lot of problems that plague other tools (character escaping, ambiguity, indentation, etc.). Additionally ytt provides Python-like language (Starlark) that executes in a hermetic environment making it friendly, yet more deterministic compared to just using general purpose languages directly or non-familiar custom templating languages.
```

Comment:
This paragraph is true. It points at a real differentiator for `ytt`.

But I can't help but consider whether it's on-topic for this article...? 🤔

Reply:
Yeah, this can be discussed. keeping this open to get thoughts from others also on this.

Reply:
> Yeah, this can be discussed but I feel there is no harm in putting what makes ytt different.

Agreed. 👍🏻